x/staking/types/msg.go

```diff
@@ -336,3 +338,58 @@ func (msg MsgUndelegate) ValidateBasic() error {
 
 	return nil
 }
+
+// NewMsgCancelUnbondingDelegation creates a new MsgCancelUnbondingDelegation instance.
+//nolint:interfacer
+func NewMsgCancelUnbondingDelegation(delAddr sdk.AccAddress, valAddr sdk.ValAddress, creationHeight uint64, amount sdk.Coin) *MsgCancelUnbondingDelegation {
+	return &MsgCancelUnbondingDelegation{
+		DelegatorAddress: delAddr.String(),
+		ValidatorAddress: valAddr.String(),
+		Amount:           amount,
+		CreationHeight:   creationHeight,
+	}
+}
+
+// Route implements the sdk.Msg interface.
+func (msg MsgCancelUnbondingDelegation) Route() string { return RouterKey }
+
+// Type implements the sdk.Msg interface.
+func (msg MsgCancelUnbondingDelegation) Type() string { return TypeMsgCancelUnbondingDelegation }
+
+// GetSigners implements the sdk.Msg interface.
+func (msg MsgCancelUnbondingDelegation) GetSigners() []sdk.AccAddress {
+	delegator, _ := sdk.AccAddressFromBech32(msg.DelegatorAddress)
+	return []sdk.AccAddress{delegator}
+}
+
+// GetSignBytes implements the sdk.Msg interface.
+func (msg MsgCancelUnbondingDelegation) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// ValidateBasic implements the sdk.Msg interface.
+func (msg MsgCancelUnbondingDelegation) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.DelegatorAddress); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid delegator address: %s", err)
+	}
+	if _, err := sdk.ValAddressFromBech32(msg.ValidatorAddress); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid validator address: %s", err)
+	}
+
+	if !msg.Amount.IsValid() || !msg.Amount.Amount.IsPositive() {
```

Comment:
is amount = 0 case covered here? need to throw error when amount = 0.

Reply:
0 isn't positive, so yes, this will throw an error when 0.